client/sites/components/site-preview-pane/preview-pane-header-buttons.tsx

```diff
@@ -54,11 +55,25 @@ const PreviewPaneHeaderButtons = ( { focusRef, itemData }: Props ) => {
 		queryClient
 	);
 
+	// Get the staging site to check its jetpack_connection
+	const stagingSiteId = isStagingSite
+		? itemData.blogId
+		: site?.options?.wpcom_staging_blog_ids?.[ 0 ];
+
+	const { data: stagingSite } = useQuery( {
+		...siteByIdQuery( stagingSiteId ?? 0 ),
+		enabled: !! stagingSiteId,
+		refetchInterval: 5000,
```

Comment:
Here we can stop polling after the `jetpack_connection` is established:

```suggestion
		refetchInterval: ( query ) => {
			return query.state.data?.jetpack_connection ? false : 5000;
		},
```

Reply:
Makes sense, let's apply this suggestion!